pkg/security/resolvers/dns/resolver.go

```diff
@@ -0,0 +1,169 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+// Package dns resolves ip addresses to hostnames
+package dns
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/security/metrics"
+	"github.com/DataDog/datadog-agent/pkg/security/probe/config"
+	"github.com/DataDog/datadog-go/v5/statsd"
+	lru "github.com/hashicorp/golang-lru/v2"
+	"go.uber.org/atomic"
+	"net/netip"
+	"slices"
+)
+
+// CacheStats defines metrics for the LRU
+type CacheStats struct {
+	cacheHits       atomic.Int64
+	cacheMisses     atomic.Int64
+	cacheInsertions atomic.Int64
+	cacheEvictions  atomic.Int64
+}
+
+// Resolver defines a DNS resolver
+type Resolver struct {
+	cache         *lru.Cache[netip.Addr, []string]
+	cnameCache    *lru.Cache[string, []string]
+	statsdClient  statsd.ClientInterface
+	resolverStats *CacheStats
+	cnameStats    *CacheStats
+}
+
+// NewDNSResolver returns a new resolver
+func NewDNSResolver(cfg *config.Config, statsdClient statsd.ClientInterface) (*Resolver, error) {
```

Comment:
Since there is no eviction mechanism (apart from the LRU algorithm), we'll always eventually hit the `cfg.DNSResolverCacheSize` max size in both LRU. Do you have a vague idea of how big this represents ? (are we talking 1 MB or 2 MB or ~10MB ?)

Also for the V2, it would be great to add a DNS response expiry mechanism, so we only keep the entries that we need.